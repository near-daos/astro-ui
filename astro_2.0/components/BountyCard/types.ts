import { Token } from 'types/token';
import { BountyStatus } from 'types/bounties';

export enum CardType {
  Bounty = 'Bounty',
  Claim = 'Claim',
}
export interface BountyCardContent {
  id: string;
  daoId: string;
  amount: string;
  type: CardType;
  description: string;
  externalUrl?: string;
  token: Token;
  status: BountyStatus;
  bountyBond: string;
  slots?: number;
  slotsTotal?: number;
  claimedByCurrentUser: boolean;
  accountId?: string;
  timeToComplete: string;
  forgivenessPeriod: string;
  relatedProposalHash?: string;
}
