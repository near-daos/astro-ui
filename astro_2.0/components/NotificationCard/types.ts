import { NotificationType, NotificationStatus } from 'types/notification';

export interface NotificationCardContent {
  id: string;
  type: NotificationType;
  status: NotificationStatus;
  text: string;
  time: string;
  flagCover?: string;
  logo?: string;
  url?: string;
}
