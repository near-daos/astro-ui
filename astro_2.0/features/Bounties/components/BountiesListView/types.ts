import { Bounty } from 'types/bounties';
import { ReactNode } from 'react';
import { ProposalVariant } from 'types/proposal';

export type SectionItem = {
  id: string;
  title: string;
  proposer: string;
  proposalId: string;
  bounty?: Bounty;
  link: {
    pathname: string;
    query: {
      dao: string;
      bounty?: string;
      proposal?: string;
    };
  };
  completeHandler?: (
    id: number,
    variant: ProposalVariant.ProposeDoneBounty
  ) => void;
  content: ReactNode;
};
