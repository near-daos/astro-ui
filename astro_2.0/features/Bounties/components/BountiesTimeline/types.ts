export type TimelineMilestoneType =
  | 'Proposal Created'
  | 'Bounty Created'
  | 'Claim'
  | 'Pending Approval'
  | 'Complete Claim'
  | 'Claim Deadline'
  | 'Complete Bounty';

export type TimelineMilestone = {
  type: TimelineMilestoneType;
  date: Date;
  tooltip: string;
  color?: string;
};

export type TimelineClaim = {
  id: string;
  milestones: TimelineMilestone[];
  title: string;
  color: string;
  minDate?: Date;
  maxDate?: Date;
};

export type TimelineGroup = {
  id: string;
  name: string;
  isOpen: boolean;
  milestones: TimelineMilestone[];
  claims: TimelineClaim[];
  minDate?: Date;
  maxDate?: Date;
};

export type TimelineGranularity = 'month' | 'day' | 'hour';
