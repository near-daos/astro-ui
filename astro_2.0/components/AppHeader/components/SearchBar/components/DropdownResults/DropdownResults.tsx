import isEmpty from 'lodash/isEmpty';
import { useRouter } from 'next/router';
import { useTranslation } from 'next-i18next';
import React, { VFC, useCallback } from 'react';
import { useWalletContext } from 'context/WalletContext';

import { SEARCH_PAGE_URL } from 'constants/routing';

import { useSearchResults } from 'features/search/search-results';
import { NoSearchResultsView } from 'features/search/search-results/components/NoSearchResultsView';

import {
  ResultSection,
  SearchResultDaoCard,
  SearchResultPeopleCard,
  SearchResultProposalLine,
} from './components/ResultSection';

import styles from './DropdownResults.module.scss';

interface DropdownResultsProps {
  width: number;
  closeSearch: () => void;
  query: string;
}

export const DropdownResults: VFC<DropdownResultsProps> = ({
  width,
  closeSearch,
  query,
}) => {
  const DAOS_TAB_IDNEX = 0;
  const PROPOSAL_TAB_INDEX = 1;
  const PEOPLE_TAB_INDEX = 2;

  const { accountId } = useWalletContext();
  const { t } = useTranslation('common');

  const router = useRouter();
  const { searchResults } = useSearchResults();

  const { daos, proposals, members } = searchResults || {};

  const goToSearchPage = useCallback(
    (tabIndex: number) => {
      router.push(
        {
          pathname: SEARCH_PAGE_URL,
          query: {
            ...router.query,
            tab: tabIndex,
          },
        },
        undefined,
        {
          shallow: true,
        }
      );
    },
    [router]
  );

  const goToDaosTabOnSearchPage = useCallback(() => {
    goToSearchPage(DAOS_TAB_IDNEX);
  }, [goToSearchPage]);

  const goToProposalsTabOnSearchPage = useCallback(() => {
    goToSearchPage(PROPOSAL_TAB_INDEX);
  }, [goToSearchPage]);

  const goToPeopleTabOnSearchPage = useCallback(() => {
    goToSearchPage(PEOPLE_TAB_INDEX);
  }, [goToSearchPage]);

  function renderNoResults() {
    if (
      isEmpty(daos) &&
      isEmpty(proposals) &&
      isEmpty(members) &&
      query !== ''
    ) {
      return <NoSearchResultsView query={query} />;
    }

    return null;
  }

  function getThreeFirstResults<T>(data?: T[]): T[] {
    return data?.slice(0, 3) || [];
  }

  function renderDaos() {
    const sorted =
      daos?.sort((a, b) => {
        if (a.accountIds.includes(accountId)) {
          return -1;
        }

        if (b.accountIds.includes(accountId)) {
          return 1;
        }

        return 0;
      }) ?? [];

    const dataToRender = getThreeFirstResults(sorted);

    return (
      <ResultSection
        data={sorted}
        title={t('header.search.daos')}
        onSeeAll={goToDaosTabOnSearchPage}
      >
        {dataToRender.map(dao => (
          <SearchResultDaoCard data={dao} key={dao.id} onClick={closeSearch} />
        ))}
      </ResultSection>
    );
  }

  function renderProposals() {
    const proposalsToRender = getThreeFirstResults(proposals);

    return (
      <ResultSection
        title={t('header.search.proposals')}
        data={proposals}
        onSeeAll={goToProposalsTabOnSearchPage}
        contentClassName={styles.proposalsContent}
      >
        {proposalsToRender.map(proposal => (
          <SearchResultProposalLine
            data={proposal}
            key={proposal.id}
            onClick={goToProposalsTabOnSearchPage}
          />
        ))}
      </ResultSection>
    );
  }

  function renderPeople() {
    const peopleToRender = getThreeFirstResults(members);

    return (
      <ResultSection
        title={t('header.search.people')}
        data={members}
        onSeeAll={goToPeopleTabOnSearchPage}
      >
        {peopleToRender.map(person => (
          <SearchResultPeopleCard
            data={person}
            key={person.id}
            onClick={goToPeopleTabOnSearchPage}
          />
        ))}
      </ResultSection>
    );
  }

  return (
    <div className={styles.root} style={{ width }}>
      {renderDaos()}
      {renderProposals()}
      {renderPeople()}

      {renderNoResults()}
    </div>
  );
};
