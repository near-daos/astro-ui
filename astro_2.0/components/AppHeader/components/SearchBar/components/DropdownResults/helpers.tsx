import styles from 'astro_2.0/components/AppHeader/components/SearchBar/components/DropdownResults/DropdownResults.module.scss';
import { Badge } from 'components/Badge';
import React, { ReactNode } from 'react';
import { Proposal, ProposalType } from 'types/proposal';
import { Button } from 'components/button/Button';
import TextTruncate from 'react-text-truncate';

export function getProposalSearchSummary(
  proposal: Proposal,
  onClick: () => void
): ReactNode {
  let content;

  switch (proposal.kind.type) {
    case ProposalType.AddMemberToRole: {
      content = (
        <div className={styles.summary}>
          <span>
            Add <strong>{proposal.kind.memberId}</strong> as a member to
          </span>
          <Badge size="small">{proposal.kind.role}</Badge>
        </div>
      );
      break;
    }
    case ProposalType.RemoveMemberFromRole: {
      content = (
        <div className={styles.summary}>
          <span>
            Remove <strong>{proposal.kind.memberId}</strong> from
          </span>
          <Badge size="small">{proposal.kind.role}</Badge>
        </div>
      );
      break;
    }
    case ProposalType.ChangePolicy: {
      content = (
        <div className={styles.summary}>
          <span>
            This is a proposal to change voting policies for our groups.
          </span>
        </div>
      );
      break;
    }
    case ProposalType.FunctionCall: {
      content = (
        <div className={styles.summary}>
          <span>Function call</span>
        </div>
      );
      break;
    }
    case ProposalType.Transfer: {
      content = (
        <div className={styles.summary}>
          <span>I would like to request a payment</span>
        </div>
      );
      break;
    }
    case ProposalType.Vote:
    case ProposalType.AddBounty:
    case ProposalType.BountyDone:
    case ProposalType.UpgradeSelf:
    case ProposalType.UpgradeRemote:
    case ProposalType.SetStakingContract:
    default: {
      content = (
        <div className={styles.summary}>
          <TextTruncate
            line={3}
            element="span"
            truncateText="â€¦"
            text={proposal.description}
            textTruncateChild={null}
          />
        </div>
      );
    }
  }

  return (
    <Button
      variant="tertiary"
      onClick={onClick}
      className={styles.row}
      key={proposal.id}
    >
      {content}
    </Button>
  );
}
