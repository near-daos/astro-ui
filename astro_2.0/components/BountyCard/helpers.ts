import { format, formatDistance, isAfter } from 'date-fns';

import {
  BountyCardContent,
  CardType,
} from 'astro_2.0/components/BountyCard/types';
import { Tokens } from 'astro_2.0/features/CustomTokens/CustomTokensContext';
import { DAO } from 'types/dao';
import { Bounty, BountyStatus, ClaimedBy } from 'types/bounties';

export const toMillis = (timePeriod: string): number =>
  Math.round(Number(timePeriod) / 1000000);

const calculateBountyStatus = (claim: ClaimedBy): BountyStatus => {
  return isAfter(Date.now(), toMillis(claim.endTime))
    ? BountyStatus.Expired
    : BountyStatus.InProgress;
};

export const getDistanceFromNow = (timePeriod: string): string => {
  return formatDistance(new Date(toMillis(timePeriod)), 0, {
    addSuffix: false,
  });
};

const canClaim = (bounty: Bounty, currentUser: string) => {
  const claimedByMe = bounty.claimedBy
    .map(({ accountId: claimedAccount }) => claimedAccount)
    .includes(currentUser);

  return bounty.slots !== 0 && !claimedByMe;
};

export const formatDate = (date: string): string => {
  return format(toMillis(date), 'd LLLL yyyy');
};

export const mapBountyToCardContent = (
  dao: DAO,
  bounty: Bounty,
  tokens: Tokens,
  currentUser: string,
  filterStatus: BountyStatus
): BountyCardContent[] => {
  const tokenData =
    bounty.tokenId === '' ? tokens.NEAR : tokens[bounty.tokenId];

  const commonProps = {
    id: bounty.id,
    daoId: dao.id,
    amount: bounty.amount,
    description: bounty.description,
    externalUrl: bounty.externalUrl,
    forgivenessPeriod: bounty.forgivenessPeriod,
    token: tokenData,
    bountyBond: dao.policy.bountyBond,
  };

  const result: BountyCardContent[] = [];

  const includeBountyStatuses = [BountyStatus.Available];
  const includeClaimsStatuses = [BountyStatus.InProgress, BountyStatus.Expired];

  if (
    (!filterStatus || includeBountyStatuses.includes(filterStatus)) &&
    canClaim(bounty, currentUser)
  ) {
    result.push({
      ...commonProps,
      status: BountyStatus.Available,
      type: CardType.Bounty,
      timeToComplete: getDistanceFromNow(bounty.deadlineThreshold),
      slots: bounty.slots,
      slotsTotal: bounty.slotsTotal,
      claimedByCurrentUser: false,
    });
  }

  if (!filterStatus || includeClaimsStatuses.includes(filterStatus)) {
    result.push(
      ...bounty.claimedBy.map(claim => {
        return {
          ...commonProps,
          type: CardType.Claim,
          status: calculateBountyStatus(claim),
          claimedBy: claim.accountId,
          claimedByCurrentUser: claim.accountId === currentUser,
          timeToComplete: formatDate(claim.endTime),
        };
      })
    );
  }

  return result;
};

export const showActionBar = (
  bountyCardContent: BountyCardContent,
  account: string | undefined
): boolean => {
  return !!account;
};
