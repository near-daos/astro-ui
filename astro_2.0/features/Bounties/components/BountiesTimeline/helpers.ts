import { BountyClaim, BountyContext, BountyProposal } from 'types/bounties';
import {
  TimelineClaim,
  TimelineGranularity,
  TimelineGroup,
  TimelineMilestone,
} from 'astro_2.0/features/Bounties/components/BountiesTimeline/types';
import { DATA_SEPARATOR } from 'constants/common';
import {
  addDays,
  eachDayOfInterval,
  eachHourOfInterval,
  eachMinuteOfInterval,
  eachMonthOfInterval,
  endOfDay,
  endOfHour,
  endOfMonth,
  format,
  getDaysInMonth,
  isLastDayOfMonth,
  isSameDay,
  isSameHour,
  startOfDay,
  startOfHour,
  startOfMinute,
  startOfMonth,
} from 'date-fns';
import { toMillis } from 'utils/format';

const TOOLTIP_DATE_FORMAT = 'dd MMM, yyyy';

const COLORS = [
  '#DC88F5',
  '#89B1F9',
  '#51BAD1',
  '#EF7F31',
  '#E43AE3',
  '#57B99D',
  '#D32C1F',
  '#89B1F9',
  '#A87F3D',
  '#76140C',
  '#D65B26',
  '#B6BCC1',
  '#EC5281',
  '#EF87AA',
  '#925EB1',
  '#8060D9',
  '#0038DF',
  '#4581B2',
  '#46A8EE',
  '#65C978',
  '#697683',
];

function getColor(index: number) {
  return COLORS[index % COLORS.length];
}

function getClaimMilestones(
  claim: BountyClaim,
  color: string,
  proposal?: BountyProposal
) {
  const { accountId, startTime } = claim;
  const milestones: TimelineMilestone[] = [];
  const claimStartTime = toMillis(startTime);
  const claimStart = new Date(claimStartTime);

  if (claimStart) {
    milestones.push({
      type: 'Claim',
      date: claimStart,
      tooltip: `Claimed by ${accountId}`,
      color,
    });
  }

  if (proposal) {
    const proposalStatus = proposal.status;

    if (proposalStatus === 'InProgress') {
      milestones.push({
        type: 'Pending Approval',
        date: new Date(proposal.createdAt),
        tooltip: 'Pending Approval',
        color,
      });
    } else if (proposalStatus === 'Approved') {
      milestones.push({
        type: 'Complete Claim',
        date: new Date(proposal.createdAt),
        tooltip: `Claim completed by ${proposal.proposer}`,
        color,
      });
    }
    // todo - how cna we handle rejected/not approved claims?
  }

  if (claimStart && claim.deadline) {
    const hasApprovedProposal = !!milestones.find(
      prp => prp.type === 'Complete Claim'
    );

    if (!hasApprovedProposal) {
      const deadline = toMillis(claim.deadline);
      const claimEnd = new Date(claimStartTime + deadline);

      milestones.push({
        type: 'Claim Deadline',
        date: claimEnd,
        tooltip: `Deadline to complete ${accountId} claim ${format(
          claimEnd,
          TOOLTIP_DATE_FORMAT
        )}`,
        color,
      });
    }
  }

  return milestones;
}

function prepareClaims(context: BountyContext) {
  const claims: TimelineClaim[] = [];

  // Bounty claims
  if (context.bounty?.bountyClaims?.length > 0) {
    context.bounty.bountyClaims.forEach((claim, i) => {
      const { id } = claim;
      const proposal = context.bounty.bountyDoneProposals.find(
        _proposal => _proposal.bountyClaimId === id
      );

      const milestones = getClaimMilestones(claim, getColor(i), proposal);

      const sortedMilestones = milestones.slice().sort((a, b) => {
        if (a.date > b.date) {
          return 1;
        }

        if (a.date < b.date) {
          return -1;
        }

        return 0;
      });

      const minDate = sortedMilestones[0]?.date;
      const maxDate = sortedMilestones[sortedMilestones.length - 1]?.date;

      claims.push({
        id: claim.id,
        milestones,
        title: claim.accountId,
        color: getColor(i),
        minDate,
        maxDate,
      });
    });
  }

  return claims;
}

function accumulateClaimsMilestones(
  milestones: TimelineMilestone[],
  claims: TimelineClaim[]
) {
  return claims.reduce(
    (res, item) => {
      res.push(...item.milestones);

      return res;
    },
    [...milestones]
  );
}

export function prepareTimelineDataset(data: BountyContext[]): TimelineGroup[] {
  return data.map(context => {
    const item = context.bounty || context.proposal;
    let milestones: TimelineMilestone[] = [];
    const [description] = item.description.split(DATA_SEPARATOR);

    if (context.proposal) {
      // Create proposal milestone
      const proposalCreated = new Date(context.proposal.createdAt);

      milestones.push({
        type: 'Proposal Created',
        date: proposalCreated,
        tooltip: `Proposal Created ${format(
          proposalCreated,
          TOOLTIP_DATE_FORMAT
        )}`,
      });
    }

    if (context.bounty) {
      // Create bounty milestone
      const bountyCreated = new Date(context.bounty.createdAt);

      milestones.push({
        type: 'Bounty Created',
        date: bountyCreated,
        tooltip: `Bounty Created ${format(bountyCreated, TOOLTIP_DATE_FORMAT)}`,
      });
    }

    if (Number(context.bounty?.times) === 0) {
      // Complete bounty milestone
      let latestDate: Date | null = null;

      context.bounty.bountyDoneProposals
        .filter(_proposal => _proposal.status === 'Approved')
        .forEach(_proposal => {
          const updatedAt = new Date(_proposal.updatedAt);

          if (!latestDate || latestDate < updatedAt) {
            latestDate = updatedAt;
          }
        });

      if (latestDate) {
        milestones.push({
          type: 'Complete Bounty',
          date: latestDate,
          tooltip: `Bounty Completed ${format(
            latestDate,
            TOOLTIP_DATE_FORMAT
          )}`,
        });
      }
    }

    const claims = prepareClaims(context);

    milestones = accumulateClaimsMilestones(milestones, claims);

    const sortedMilestones = milestones.slice().sort((a, b) => {
      if (a.date > b.date) {
        return 1;
      }

      if (a.date < b.date) {
        return -1;
      }

      return 0;
    });

    const groupMinDate = sortedMilestones[0]?.date;
    const groupMaxDate = sortedMilestones[sortedMilestones.length - 1]?.date;

    return {
      id: context.id,
      isOpen: false,
      name: description || '',
      milestones,
      claims,
      minDate: groupMinDate,
      maxDate: groupMaxDate,
    };
  });
}

export function getTimelineRange(dataset: TimelineGroup[]): [Date, Date] {
  let min: Date | null = null; // new Date();
  let max: Date | null = null; // addMonths(min, 1);

  dataset.forEach(group => {
    group.milestones.forEach(milestone => {
      if (!min || min > milestone.date) {
        min = milestone.date;
      }

      if (!max || max < milestone.date) {
        max = milestone.date;
      }
    });
  });

  if (!min) {
    min = new Date();
  }

  if (!max) {
    max = addDays(min, 2);
  }

  return [min, max];
}

export function getTopColumns(
  start: Date,
  end: Date,
  granularity: TimelineGranularity
): Date[] {
  switch (granularity) {
    case 'hour': {
      return eachHourOfInterval({ start, end });
    }
    case 'day': {
      return eachDayOfInterval({ start, end });
    }
    default: {
      return eachMonthOfInterval({ start, end });
    }
  }
}

export function getColumns(
  start: Date,
  end: Date,
  granularity: TimelineGranularity
): Date[] {
  switch (granularity) {
    case 'hour': {
      return eachMinuteOfInterval({ start, end });
    }
    case 'day': {
      return eachHourOfInterval({ start, end });
    }
    default: {
      return eachDayOfInterval({ start, end });
    }
  }
}

export function formatColumnLabel(
  date: Date,
  granularity: TimelineGranularity
): string {
  switch (granularity) {
    case 'hour': {
      return format(date, 'mm');
    }
    case 'day': {
      return format(date, 'H');
    }
    default: {
      return format(date, 'd');
    }
  }
}

export function getTopColumnsWidth(
  date: Date,
  granularity: TimelineGranularity
): number {
  switch (granularity) {
    case 'hour': {
      return 60 * 32;
    }
    case 'day': {
      return 24 * 32;
    }
    default: {
      return getDaysInMonth(date) * 32;
    }
  }
}

export function getRangeColumns(
  date: Date,
  granularity: TimelineGranularity
): Date[] {
  switch (granularity) {
    case 'hour': {
      return eachMinuteOfInterval({
        start: startOfHour(date),
        end: endOfHour(date),
      });
    }
    case 'day': {
      return eachHourOfInterval({
        start: startOfDay(date),
        end: endOfDay(date),
      });
    }
    default: {
      return eachDayOfInterval({
        start: startOfMonth(date),
        end: endOfMonth(date),
      });
    }
  }
}

export function getMilestonesForDate(
  milestones: TimelineMilestone[],
  date: Date,
  granularity: TimelineGranularity
): TimelineMilestone[] {
  let compareFn = isSameDay;

  switch (granularity) {
    case 'day': {
      compareFn = isSameHour;

      break;
    }
    default: {
      compareFn = isSameDay;

      break;
    }
  }

  return (
    milestones?.filter(item => {
      return compareFn(date, item.date);
    }) ?? []
  );
}

export function getGroupDateStart(
  date: Date | undefined,
  granularity: TimelineGranularity
): Date | null {
  if (!date) {
    return null;
  }

  switch (granularity) {
    case 'hour': {
      return startOfMinute(date);
    }
    case 'day': {
      return startOfHour(date);
    }
    default: {
      return startOfDay(date);
    }
  }
}

export function isEndOfGranularityPeriod(
  date: Date,
  granularity: TimelineGranularity
): boolean {
  switch (granularity) {
    case 'hour': {
      return date.getMinutes() === 59;
    }
    case 'day': {
      return date.getHours() === 23;
    }
    default: {
      return isLastDayOfMonth(date);
    }
  }
}
