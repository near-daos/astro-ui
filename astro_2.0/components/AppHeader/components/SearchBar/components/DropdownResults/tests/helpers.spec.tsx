import { Proposal, ProposalType } from 'types/proposal';

import { getProposalSearchSummary } from 'astro_2.0/components/AppHeader/components/SearchBar/components/DropdownResults/helpers';

describe('search results dropdown helpers', () => {
  it.each`
    type
    ${ProposalType.AddMemberToRole}
    ${ProposalType.RemoveMemberFromRole}
    ${ProposalType.ChangePolicy}
    ${ProposalType.FunctionCall}
    ${ProposalType.Transfer}
    ${ProposalType.Vote}
    ${ProposalType.AddBounty}
    ${ProposalType.BountyDone}
    ${ProposalType.UpgradeSelf}
    ${ProposalType.UpgradeRemote}
    ${ProposalType.SetStakingContract}
  `('Should render proper content for $type proposals', ({ type }) => {
    const proposal = {
      kind: {
        type,
      },
    } as Proposal;

    // eslint-disable-next-line @typescript-eslint/no-empty-function
    const component = getProposalSearchSummary(proposal, () => {});

    expect(component).toMatchSnapshot();
  });
});
