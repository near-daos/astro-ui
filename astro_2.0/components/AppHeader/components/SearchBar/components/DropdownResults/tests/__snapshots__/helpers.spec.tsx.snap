// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`search results dropdown helpers Should render proper content for AddBounty proposals 1`] = `
<Button
  className="row"
  onClick={[Function]}
  variant="tertiary"
>
  <div
    className="summary"
  >
    <TextTruncate
      element="span"
      line={3}
      maxCalculateTimes={10}
      text=""
      textElement="span"
      textTruncateChild={null}
      truncateText="…"
    />
  </div>
</Button>
`;

exports[`search results dropdown helpers Should render proper content for AddMemberToRole proposals 1`] = `
<Button
  className="row"
  onClick={[Function]}
  variant="tertiary"
>
  <div
    className="summary"
  >
    <span>
      Add 
      <strong />
       as a member to
    </span>
    <ForwardRef
      size="small"
    />
  </div>
</Button>
`;

exports[`search results dropdown helpers Should render proper content for BountyDone proposals 1`] = `
<Button
  className="row"
  onClick={[Function]}
  variant="tertiary"
>
  <div
    className="summary"
  >
    <TextTruncate
      element="span"
      line={3}
      maxCalculateTimes={10}
      text=""
      textElement="span"
      textTruncateChild={null}
      truncateText="…"
    />
  </div>
</Button>
`;

exports[`search results dropdown helpers Should render proper content for ChangePolicy proposals 1`] = `
<Button
  className="row"
  onClick={[Function]}
  variant="tertiary"
>
  <div
    className="summary"
  >
    <span>
      This is a proposal to change voting policies for our groups.
    </span>
  </div>
</Button>
`;

exports[`search results dropdown helpers Should render proper content for FunctionCall proposals 1`] = `
<Button
  className="row"
  onClick={[Function]}
  variant="tertiary"
>
  <div
    className="summary"
  >
    <span>
      Function call
    </span>
  </div>
</Button>
`;

exports[`search results dropdown helpers Should render proper content for RemoveMemberFromRole proposals 1`] = `
<Button
  className="row"
  onClick={[Function]}
  variant="tertiary"
>
  <div
    className="summary"
  >
    <span>
      Remove 
      <strong />
       from
    </span>
    <ForwardRef
      size="small"
    />
  </div>
</Button>
`;

exports[`search results dropdown helpers Should render proper content for SetStakingContract proposals 1`] = `
<Button
  className="row"
  onClick={[Function]}
  variant="tertiary"
>
  <div
    className="summary"
  >
    <TextTruncate
      element="span"
      line={3}
      maxCalculateTimes={10}
      text=""
      textElement="span"
      textTruncateChild={null}
      truncateText="…"
    />
  </div>
</Button>
`;

exports[`search results dropdown helpers Should render proper content for Transfer proposals 1`] = `
<Button
  className="row"
  onClick={[Function]}
  variant="tertiary"
>
  <div
    className="summary"
  >
    <span>
      I would like to request a payment
    </span>
  </div>
</Button>
`;

exports[`search results dropdown helpers Should render proper content for UpgradeRemote proposals 1`] = `
<Button
  className="row"
  onClick={[Function]}
  variant="tertiary"
>
  <div
    className="summary"
  >
    <TextTruncate
      element="span"
      line={3}
      maxCalculateTimes={10}
      text=""
      textElement="span"
      textTruncateChild={null}
      truncateText="…"
    />
  </div>
</Button>
`;

exports[`search results dropdown helpers Should render proper content for UpgradeSelf proposals 1`] = `
<Button
  className="row"
  onClick={[Function]}
  variant="tertiary"
>
  <div
    className="summary"
  >
    <TextTruncate
      element="span"
      line={3}
      maxCalculateTimes={10}
      text=""
      textElement="span"
      textTruncateChild={null}
      truncateText="…"
    />
  </div>
</Button>
`;

exports[`search results dropdown helpers Should render proper content for Vote proposals 1`] = `
<Button
  className="row"
  onClick={[Function]}
  variant="tertiary"
>
  <div
    className="summary"
  >
    <TextTruncate
      element="span"
      line={3}
      maxCalculateTimes={10}
      text=""
      textElement="span"
      textTruncateChild={null}
      truncateText="…"
    />
  </div>
</Button>
`;
